### DISCUSSION TITLE
I2C Communication not working in STM32

### DISCUSSION THREAD

**Discussion by DIEGGOMEZ** [Infineon/optiga-trust-m#107] [Q&A]
Good morning,

I have migrate Optiga Trust M library to Keil uVision, my idea is to use Optiga Trust M to read the default certificate (0xE0E0), but i am just reading 0x00, and no I2C activity is going on.

This is my code: 
![image](https://github.com/Infineon/optiga-trust-m/assets/62071761/1ecec59d-eeb7-4e4b-948d-afd07e70f03a)
 
If i use HAL I2C directly, Optiga is answering me, but if i use the optiga libraries no answer is happening,
Any ideas why I2C is not working?

Thanks in advance.

**Comment by ShaikKarishma0** [comment]
Please refer the following link - https://github.com/Infineon/optiga-trust-m/issues/108
